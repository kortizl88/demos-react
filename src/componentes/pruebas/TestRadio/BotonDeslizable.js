import React from 'react'

export const BotonDeslizable = ({ leyenda, inactivo = false, estado,  eventoCambioEstado }) => {
    return (
        <>
            <label>{leyenda}</label>
            <input 
                type='checkbox'
                checked={estado}
                disabled={inactivo}
                onChange={(e) => eventoCambioEstado(e.target.checked)}
            />
        </>
    )
}
