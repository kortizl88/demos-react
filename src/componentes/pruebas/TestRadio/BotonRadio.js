import React from 'react'

export const BotonRadio = ({leyenda, inactivo=false, valor, estado, eventoCambioEstado}) => {
    return (
        <>
            <label>{leyenda}</label>
            <input type="radio"
                    checked={ estado === valor }
                    disabled={inactivo}
                    value={valor}
                    onChange={(e) => eventoCambioEstado(e.target.value)}
                    />  
        </>
    )
}
