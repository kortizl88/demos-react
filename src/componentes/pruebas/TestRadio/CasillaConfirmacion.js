import React from 'react'
import './CasillaConfirmacion.scss';

export const CasillaConfirmacion = ({estado, eventoCambio, identificador, texto, inactivo=false }) => {
    
    const estiloEtiqueta = () => {
        if(inactivo){
            return 'etiqueta inactivo'
        }else{
            return 'etiqueta';
        }
    };

    return (
        <div className='casilla-confirmacion'>
            <label className={estiloEtiqueta()} htmlFor={identificador} >{texto}
                <input 
                    className='control' 
                    type="checkbox" 
                    id={identificador}
                    disabled={inactivo}
                    checked={estado}
                    onChange={()=> eventoCambio(!estado)}/>
                <span className="indicador"/>
            </label>
        </div>
    )
}
