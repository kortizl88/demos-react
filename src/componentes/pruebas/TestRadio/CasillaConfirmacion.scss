.casilla-confirmacion{
    .etiqueta{
        position: relative;
        cursor: pointer;
        font-size: 14px;
        line-height: 14px;
        padding-left: 20px;
        &.inactivo{
            color:#CCCDCE;
        }
        input.control {
            position: absolute;
            opacity: 0;
            cursor: pointer;
            height: 0;
            width: 0;
          }
        .indicador{
            position: absolute;
            top: 1px;
            left: 0;
            height: 12px;
            width: 12px;
            margin-right: 8px;
            background-color: #eee;
            border: 1px solid black;
            &:after {
                content: "";
                position: absolute;
                display: none;
              }
        }

        input.control:checked ~ .indicador {
            background-color: #006341;
        }

        input.control:checked ~ .indicador:after {
            display: block;
        }

        .indicador:after {
            left: 3px;
            top: 0px;
            width: 4px;
            height: 8px;
            border: solid white;
            border-width: 0 2px 2px 0;
            transform: rotate(35deg);
          }

        input.control:disabled ~ .indicador {
            background-color: #fff;
            border-color:#CCCDCE;
        }

    }
}