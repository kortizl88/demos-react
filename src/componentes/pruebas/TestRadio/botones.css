.control {
    font-family: arial;
    display: block;
    position: relative;
    padding-left: 30px;
    margin-bottom: 0px;
    padding-top: 0px;
    cursor: pointer;
    font-size: 16px;
}
    .control input {
        position: absolute;
        z-index: -1;
        opacity: 0;
    }
.control_indicator {
    position: absolute;
    top: 1px;
    left: 0;
    height: 20px;
    width: 20px;
    background: #ffffff;
    border: 1px solid #266b19;
    border-radius: undefinedpx;
}
.control:hover input ~ .control_indicator,
.control input:focus ~ .control_indicator {
    background: #ffffff;
}

.control input:checked ~ .control_indicator {
    background: #ffffff;
}
.control:hover input:not([disabled]):checked ~ .control_indicator,
.control input:checked:focus ~ .control_indicator {
    background: #ffffff;
}
.control input:disabled ~ .control_indicator {
    background: #e6e6e6;
    opacity: 1;
    pointer-events: none;
}
.control_indicator:after {
    box-sizing: unset;
    content: '';
    position: absolute;
    display: none;
}
.control input:checked ~ .control_indicator:after {
    display: block;
}
.control-radio .control_indicator {
    border-radius: 50%;
}

.control-radio .control_indicator:after {
    left: 2px;
    top: 2px;
    height: 14px;
    width: 14px;
    border-radius: 50%;
    background: #38662f;
    transition: background 250ms;
}
.control-radio input:disabled ~ .control_indicator:after {
    background: #7b7b7b;
}