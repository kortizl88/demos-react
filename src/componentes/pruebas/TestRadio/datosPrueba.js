export const menuCat = [
    {
        id: 1,
        descripcion: 'TV',
        idPadre: 0,
        nivel: 1,
    },
    {
        id: 2,
        descripcion: 'LB',
        idPadre: 0,
        nivel: 1,
    },
    {
        id: 3,
        descripcion: 'OTRO',
        idPadre: 0,
        nivel: 1,
    },
    {
        id: 4,
        descripcion: 'ETC',
        idPadre: 0,
        nivel: 1,
    }
]