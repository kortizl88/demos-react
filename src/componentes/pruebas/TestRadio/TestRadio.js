import React, {useState} from 'react'
import { BotonDeslizable } from './BotonDeslizable';
import { BotonRadio } from './BotonRadio';
import './botones.css'

export const TestRadio = () => {

    
    const [isChecked, setIsChecked] = useState(false);
    const [radio, setRadio] = useState();

    return (
        <div>
            <h1>Pruebas de radios con useState</h1>

            <h2>Check esta checado? : { isChecked ? 'true' : 'false' } </h2>

            <h2>Radio seleccionado : { radio } </h2>

            <form>
                <h3>Check</h3>
                <BotonDeslizable 
                    leyenda='mi check personalizado'
                    estado={isChecked}
                    eventoCambioEstado={setIsChecked}
                    />

                <br/>
                <h3>Radios</h3>
                <BotonRadio 
                    leyenda="Manzana"
                    valor='manzana'
                    estado={radio}
                    eventoCambioEstado={setRadio} />
                <br/>
                <BotonRadio 
                    leyenda="Pera"
                    valor='pera'
                    estado={radio}
                    eventoCambioEstado={setRadio} />
                <br/>
                <BotonRadio 
                    leyenda="Platano"
                    valor='platano'
                    estado={radio}
                    eventoCambioEstado={setRadio} />

                    <br/>
                    <br/>
                    <br/>
                    <div class="control-group">
                        <label class="control control-radio">
                            First radio
                            <input type="radio" name="radio" checked="checked" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Second radio
                            <input type="radio" name="radio" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Disabled
                            <input type="radio" name="radio2" disabled="disabled" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Disabled & checked
                            <input type="radio" name="radio2" disabled="disabled" checked="checked" />
                            <div class="control_indicator"></div>
                        </label>
                    </div>

            </form>
        </div>
    )
}
