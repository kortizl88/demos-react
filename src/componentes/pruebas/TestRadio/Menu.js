import React from 'react'
import { menuCat } from './datosPrueba';
import { SubMenu } from './SubMenu';

export const Menu = () => {
    const menu = menuCat;
    
    const opciones = (nivel, idPadre) => menu.filter( opcion =>  opcion.nivel === nivel && opcion.idPadre === idPadre );

    return (
        <div>
            {
                opciones(1,0).map( opcion => <div className='opcion' nivel={opcion.nivel} idPadre={opcion.idPadre} >{opcion.descripcion}</div>)
            }
        </div>
    )
}
