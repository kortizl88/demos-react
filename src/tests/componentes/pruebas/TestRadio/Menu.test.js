import React from "react";
import { mount } from "enzyme";
import { Menu } from "../../../../componentes/pruebas/TestRadio/Menu";

describe('Pruebas en <Menu />', () => {
    const componente = mount(<Menu/>);

    test('Debe mostrar el menu', () => {
        expect(componente).toMatchSnapshot();
        console.log(componente.html());
        console.log(componente.findWhere(elemento => { console.log(elemento.props()); return elemento.props().nivel === '1' } ));
        expect(componente.findWhere(elemento =>  elemento.props().nivel === 1 && elemento.props().idPadre === 0 ).length).toBe(4);
        expect(componente.find('div.opcion').at(0).props().nivel === 1 && componente.find('div.opcion').at(0).props().idPadre === 0).toBe(true);
        expect(componente.find('div.opcion').at(0).props().nivel === 0 && componente.find('div.opcion').at(0).props().idPadre === 0).toBe(false);
        expect(componente.find('div.opcion').length).toBe(4);
        /*expect(componente.findWhere(elemento =>  elemento.props().nivel === 1 && elemento.props().idPadre === 1 ).exists()).toBe(false);
        expect(componente.findWhere(elemento =>  elemento.props().nivel === 0 && elemento.props().idPadre === 0).exists()).toBe(false);*/
    });


    
})
