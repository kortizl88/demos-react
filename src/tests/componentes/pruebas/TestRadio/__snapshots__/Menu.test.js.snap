// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Menu /> Debe mostrar el menu 1`] = `ReactWrapper {}`;
